odoo/modules/registry.py

```diff
@@ -137,8 +137,7 @@ def init(self, db_name):
         self.cache_sequence = None
 
         # Flags indicating invalidation of the registry or the cache.
-        self.registry_invalidated = False
-        self.cache_invalidated = False
+        self._invalidate_flags = threading.local()
```

Comment:
Would probably be a good idea to use `werkzeug.local.Local` here to avoid bad surprises in the evented worker (especially if we start using it a lot more due to websocket communications).

Also maybe `invalidation_flags` instead, to match the comment?

Reply:
Actually the doc of gevent explicitly lists `threading.local` in the patched builtins (https://www.gevent.org/api/gevent.monkey.html#gevent.monkey.patch_thread).
We already use `threading.RLock` and `threading.current_thread` BTW.